Zotero/Scenes/Detail/PDF/PDFCoordinator.swift

```diff
@@ -285,6 +296,134 @@ extension PDFCoordinator: PdfReaderCoordinatorDelegate {
+            guard let self else {
+                elementProvider([])
+                return
+            }
+            let annotationPreviewController = self.controllers.annotationPreviewController
+            let pageIndex: PageIndex = UInt(annotation.page)
+            let rect = annotation.boundingBox(boundingBoxConverter: boundingBoxConverter)
+            var size = rect.size
+            size.width *= scale
+            size.height *= scale
+            annotationPreviewController.render(
+                document: state.document,
+                page: pageIndex,
+                rect: rect,
+                imageSize: size,
+                imageScale: 1.0,
+                key: annotation.key,
+                parentKey: state.key,
+                libraryId: state.library.id
+            )
+            .observe(on: MainScheduler.instance)
+            .subscribe { [weak self] (image: UIImage) in
+                var menuTitle = title
+                // By default UIActivityViewController shares a JPEG image with 0.8 compression quality,
+                // so we compute the image size as such. Actual image produced can be overriden if we need to.
+                if let imageData = image.jpegData(compressionQuality: 0.8) {
+                    let sizeInBytes = imageData.count
+                    let sizeInKB = Double(sizeInBytes) / 1024.0
+                    if sizeInKB < 1024.0 {
+                        menuTitle += " (\(String(format: "%.0f", sizeInKB)) KB)"
+                    } else {
+                        let sizeInMB = sizeInKB / 1024.0
+                        menuTitle += " (\(String(format: "%.2f", sizeInMB)) MB)"
+                    }
+                }
+                let action = UIAction(title: menuTitle) { [weak self] (_: UIAction) in
+                    guard let self else { return }
+                    DDLogInfo("PDFCoordinator: share pdf annotation image - \(title)")
+                    if let coordinator = self.childCoordinators.last, coordinator is AnnotationPopoverCoordinator {
+                        coordinator.share(item: image, sourceView: .view(sender, nil))
+                    } else {
+                        (self as Coordinator).share(item: image, sourceView: .view(sender, nil))
+                    }
+                }
+                action.accessibilityLabel = L10n.Accessibility.Pdf.shareAnnotationImage + " " + title
+                action.isAccessibilityElement = true
+                elementProvider([action])
+            } onFailure: { (error: Error) in
+                DDLogError("PDFCoordinator: can't render annotation image - \(error)")
+                elementProvider([])
+            }
+            .disposed(by: disposeBag)
+        }
+    }
+        
+    func shareAnnotationMenu(
+        state: PDFReaderState,
+        annotation: Annotation,
+        sender: UIButton
+    ) -> UIMenu? {
+        var children: [UIMenuElement] = []
```

Comment:
I'd probably change this to:

```
guard annotation.type == .image, let boundingBoxConverter = self.navigationController.viewControllers.last as? AnnotationBoundingBoxConverter else { return nil }

...

return UIMenu(children: children)
```

Since in the end you have to check for `guard !children.isEmpty else { return nil }` anyway and currently they are only empty if the initial condition fails.

Reply:
The reasoning is that as we add more sharing sections, each one may add additional children, independently. E.g. for text, comment, etc.

Reply:
Check https://github.com/zotero/zotero-ios/pull/693/commits/f5b4428b61f55a2daa5dc429ef48463b6f37845e